### PR TITLE
refactor: convert exploreresultsbutton from class component to functional component

### PR DESCRIPTION
**Pull request by jcahela**
### SUMMARY

I converted the ExploreResultsButton from class component to functional component.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx [hunk 1]
```diff
@@ -60,57 +52,55 @@ class ExploreResultsButton extends React.PureComponent {
     return [];
   }
 
-  getQueryDuration() {
+  function getQueryDuration() {
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx:55]: Please convert this to an arrow function.

File: superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx [hunk 2]
```diff
@@ -60,57 +52,55 @@ class ExploreResultsButton extends React.PureComponent {
...
     return moment
-      .duration(this.props.query.endDttm - this.props.query.startDttm)
+      .duration(props.query.endDttm - props.query.startDttm)
       .asSeconds();
   }
 
-  getInvalidColumns() {
+  function getInvalidColumns() {
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx:61]: Please convert this to an arrow function.

File: superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx [hunk 3]
```diff
@@ -60,57 +52,55 @@ class ExploreResultsButton extends React.PureComponent {
...
-
-    return this.props.query.results.selected_columns
+    return props.query.results.selected_columns
       .map(col => col.name)
       .filter(col => re1.test(col) || re2.test(col));
   }
 
-  datasourceName() {
-    const { query } = this.props;
+  function datasourceName() {
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx:69]: Please convert this to an arrow function.

File: superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx [hunk 4]
```diff
@@ -60,57 +52,55 @@ class ExploreResultsButton extends React.PureComponent {
...
+    if (props.query) {
+      datasourceName = props.query.user ? `${props.query.user}-` : '';
+      datasourceName += `${props.query.tab}-${uniqueId}`;
     }
     return datasourceName;
   }
 
-  buildVizOptions() {
-    const { schema, sql, dbId, templateParams } = this.props.query;
+  function buildVizOptions() {
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx:79]: Please convert this to an arrow function.

Also - GH actions says that this function is defined but never used. If you come across any unused functions like this, please remove them.

File: superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx [hunk 5]
```diff
@@ -60,57 +52,55 @@ class ExploreResultsButton extends React.PureComponent {
...
       templateParams,
-      datasourceName: this.datasourceName(),
-      columns: this.getColumns(),
+      datasourceName: datasourceName(),
+      columns: getColumns(),
     };
   }
 
-  renderTimeoutWarning() {
+  function renderTimeoutWarning() {
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx:91]: Please convert this to an arrow function.

Also - GH actions says that this function is defined but never used. If you come across any unused functions like this, please remove them.

File: superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx [hunk 6]
```diff
@@ -129,8 +119,8 @@ class ExploreResultsButton extends React.PureComponent {
     );
   }
 
-  renderInvalidColumnMessage() {
-    const invalidColumns = this.getInvalidColumns();
+  function renderInvalidColumnMessage() {
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx:122]: Please convert this to an arrow function.

Also - GH actions says that this function is defined but never used. If you come across any unused functions like this, please remove them.

File: superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx [hunk 7]
```diff
@@ -149,28 +139,29 @@ class ExploreResultsButton extends React.PureComponent {
...
-            icon="line-chart"
-            placement="top"
-            label="explore"
-          />{' '}
-          {t('Explore')}
-        </Button>
-      </>
-    );
-  }
+  const allowsSubquery = props.database && props.database.allows_subquery;
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx:142]:
```suggestion
  const allowsSubquery = props.database?.allows_subquery;
```
Nit: This code can be cleaned with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).

File: superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx [hunk 8]
```diff
@@ -149,28 +139,29 @@ class ExploreResultsButton extends React.PureComponent {
...
+
+  return (
+    <>
+      <Button
+        buttonSize="small"
+        onClick={props.onClick}
+        disabled={!allowsSubquery}
+        tooltip={t(
+          'Explore the result set in the data exploration viewwwwwwwwwwwwww',
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx:151]:
```suggestion
          'Explore the result set in the data exploration view',
```

File: superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx [hunk 9]
```diff
@@ -149,28 +139,29 @@ class ExploreResultsButton extends React.PureComponent {
...
+        )}
+      >
+        <InfoTooltipWithTrigger
+          icon="line-chart"
+          placement="top"
+          label="explore"
+        />{' '}
+        {t('Jason')}
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx:159]:
```suggestion
        {t('Explore')}
```

File: superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx [hunk 10]
```diff
@@ -40,16 +40,8 @@ const defaultProps = {
...
-    super(props);
-    this.getInvalidColumns = this.getInvalidColumns.bind(this);
-    this.renderInvalidColumnMessage =
-      this.renderInvalidColumnMessage.bind(this);
-  }
-
-  getColumns() {
-    const { props } = this;
+function ExploreResultsButton(props) {
+  function getColumns() {
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx:44]: Please convert this to an arrow function.

File: superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx [hunk 11]
```diff
@@ -40,16 +40,8 @@ const defaultProps = {
...
-  constructor(props) {
-    super(props);
-    this.getInvalidColumns = this.getInvalidColumns.bind(this);
-    this.renderInvalidColumnMessage =
-      this.renderInvalidColumnMessage.bind(this);
-  }
-
-  getColumns() {
-    const { props } = this;
+function ExploreResultsButton(props) {
```
Comment (lyndsiWilliams) [inline superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx:43]: Please convert all of your functions to arrow functions (I'll mark the others with comments)

**Review by lyndsiWilliams — CHANGES_REQUESTED** [review]
Found a few changes needed here, let me know if you have any questions! 😁

**Review by geido — COMMENTED** [review]
Hey @jcahela thanks for contributing! In addition to what Lyndsi suggested here, I think it would be great to convert it to typescript as well. Let me know if you'd like to do it in this PR or another. I am happy to help you with that!

File: superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx (hunk 11 above)
Comment (AAfghahi) [reply superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx:43]: All the others I completely agree with, but I think that this, being the functional component itself, does not need to be an arrow function. 

However, I think that you need to destructure props here, you can do that by console logging the props and writing them out individually, its more explicit.

**Pull request closed without merging**